src/feature-extraction/neighbors_features.jl

```diff
@@ -13,487 +13,169 @@ struct NeighborLongitudinalResult
-    targetpoint_valid::VehicleTargetPoint;
-    max_distance_fore::Float64 = 250.0
-    ) where {S,D<:AbstractAgentDefinition,I}
-
-Returns the information about the rear neighbor of a vehicle on its own lane. 
+    findneighbor(scene::Frame, roadway::Roawday, ego::Entity; kwargs...)
+
+Search through lanes and road segments to find a neighbor of `ego` in the `scene`. 
 
 # Arguments
 
-- `scene` the scene containing all vehicles information 
-- `vehicle_index` the index of the vehicle in the scene for which we want to find the neighbor (ego vehicle)
-- `roadway` the roadway layout 
-- `targetpoint_ego` a target point on the ego vehicle that will be used to calculate the distance 
-- `targetpoint_primary` the reference point whose distance we want to minimize
-- `targetpoint_valid` the reference point, which if distance to is positive, we include the vehicle
-- `max_distance_fore` max distance to search forward [m]
-"""
-function get_neighbor_rear_along_lane(
-    scene::EntityFrame{S,D,I},
-    roadway::Roadway,
-    tag_start::LaneTag,
-    s_base::Float64,
-    targetpoint_primary::VehicleTargetPoint, # the reference point whose distance we want to minimize
-    targetpoint_valid::VehicleTargetPoint; # the reference point, which if distance to is positive, we include the vehicle
-    max_distance_rear::Float64 = 250.0, # max distance to search rearward [m]
-    index_to_ignore::Int=-1,
-    ) where {S,D<:AbstractAgentDefinition,I}
-
-    best_ind = nothing
-    best_dist = max_distance_rear
-    tag_target = tag_start
+- `scene::Frame` the scene containing all the entities
+- `roadway::Roadway` the road topology on which entities are driving 
+- `ego::Entity` the entity that we want to compute the neighbor of.
 
-    ignore = Set{I}()
+# Keyword arguments
 
+- `rear::Bool = false` set to true to search for rear neighbor, search forward neighbor by default 
+- `max_distance::Float64 = 250.0` stop searching after this distance is reached, if the neighbor is further than max_distance, returns nothing
+- `targetpoint_ego::VehicleTargetPoint` the point on the ego vehicle used for distance calculation, see `VehicleTargetPoint` for more info
+- `targetpoint_neighbor::VehicleTargetPoint` the point on the neighbor vehicle used for distance calculation, see `VehicleTargetPoint` for more info
+- `ids_to_ignore::Union{Nothing, Set{I}} = nothing` a list of entity ids to ignore for the search, ego is always ignored.
+"""
+function findneighbor(scene::Frame, roadway::Roadway, ego::Entity{S,D,I};
+                       lane::Union{Nothing, Lane} = get_lane(roadway, ego),
+                       rear::Bool=false, 
+                       max_distance::Float64=250.0, 
+                       targetpoint_ego::VehicleTargetPoint = VehicleTargetPointCenter(),
+                       targetpoint_neighbor::VehicleTargetPoint = VehicleTargetPointCenter(),
+                       ids_to_ignore::Union{Nothing, Set{I}} = nothing) where {S,D,I}
```

Comment:
Great, this is much nicer than the old functions!!
In terms of function naming, I think `find_neighbor` or even just `neighbor` would be better
https://github.com/invenia/BlueStyle#function-naming . It would help readability here and more importantly it is more consistent with existing function names in the package.

Reply:
I am quite biased on this, I know my personal opinion is in disagreement with https://docs.julialang.org/en/v1/manual/variables/index.html#Stylistic-Conventions-1 which "discourages" underscores. https://github.com/invenia/BlueStyle#function-naming seems to be more open to using underscores and I personally think omitting word spacing is a bad idea.
We can absolutely keep `findneighbor` but I wanted to give my five cents on this.

Reply:
Yes I am not sure about this either, I started with `find_neighbor` but then I thought of `findfirst`...

Reply:
I like `find_neighbor` better than just `neighbor` because it reminds that a search is happening and that it is more expensive than just `ego.neighbor`.

Reply:
replaced by find_neighbor